apps/api/src/auth/auth.controller.ts

```diff
@@ -48,6 +49,23 @@ export class AuthController {
     }
   }
 
+  @Post('/google/idtoken')
+  async authWithIdToken(
+    @Body() body: { idToken: string },
+    @Res({ passthrough: true }) res: Response
+  ) {
+    const { idToken } = body
+    const payload = await this.authService.validateGoogleIdToken(idToken)
+    if (!payload) throw new BadRequestException('Invalid id token')
+
+    const { refreshToken } = await this.authService.handleGoogleIdToken(payload)
+    res.cookie('refreshtoken', refreshToken, {
+      httpOnly: true,
+      maxAge: 1000 * 60 * 60 * 24 * 30 * 6,
```

Comment:
maybe extract this as a constant since it shares the same value as the function below